internal/tiltfile/starkit/path.go

```diff
@@ -1,18 +1,25 @@
 package starkit
 
 import (
+	"fmt"
 	"path/filepath"
+	"strings"
 
 	"go.starlark.net/starlark"
 )
 
 // We want to resolve paths relative to the dir where the currently executing file lives,
 // not relative to the working directory.
-func AbsPath(t *starlark.Thread, path string) string {
+func AbsPath(t *starlark.Thread, path string) (string, error) {
+	if strings.HasPrefix(path, "~/") || path == "~" {
+		return "", fmt.Errorf(
+			"Tiltfile does not support expansion of '~', please provide an absolute path (you provided: %q)", path)
```

Comment:
I feel like, as a user, I'd have these thoughts:
1. "You already wrote the code to detect it, it's barely any more work to do the expansion for me. Why don't you?"
2. "I don't *have* an absolute path - if I did, I would have used it"
3. "Why can't I provide a relative path?"

I think (3) is "you can" and that part of this error message is just wrong (IIUC this function takes a potentially relative path and turns it into an absolute path)
I don't really have a good answer for (1), and I think we're kind of inconsistent here - if we disallow `~` expansion, we should probably also disallow accessing files outside of the Tiltfile root. I'm not sure how to communicate this. Maybe we just ignore this.

Dan suggested users can work around with with `os.environ`, which we could technically suggest here, but then it's weird to disallow ~ expansion while encouraging `os.environ`, which is no better.

Maybe those were all wasted words because this error is probably still a net improvement.

Also, `~`-expansion *is* supported in `local` and `run`, which this error could be seen as contradicting.

I also think "Tiltfile" as a subject without "The" feels weird, but I don't know if that's just me.

Maybe "Sorry! The Tiltfile does not support ~-expansion in this context (from %q)"?

Sorry for getting nitty.

Reply:
Nah, not nitty, super useful. DB said some similar stuff irl, esp re: `~` being supported elsewhere, and us being inconsistent about our error messaging. I'll just abandon this PR